### PR TITLE
Add support for Ed25519 certificates

### PR DESCRIPTION
**Pull request by shahn**
This adds support for verification of ed25519 certificates according to
RFC 8410. Implements #49.

The test certificate was generated using OpenSSL 1.1.1a, using the
following commands (CA.pl is distributed with OpenSSL):

# CA:
openssl genpkey -algorithm ed25519 -outform pem -out root_key.pem
openssl req -new -x509 -days 9999 -extensions v3_ca -key root_key.pem \
-inform pem -outform pem -out root_ed25519.pem
echo root_ed25519.pem | CA.pl -newca

# Client:
openssl genpkey -algorithm ed25519 -outform pem -out client_key.pem
openssl req -new -key client_key.pem -inform pem -outform pem \
-out client_ed25519_csr.pem
openssl ca -keyfile ./root_key.pem -days 999 -notext -in \
client_ed25519_csr.pem -out client_ed25519.pem

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -81,6 +82,30 @@ pub fn netflix()
...
 
+#[cfg(feature = "trust_anchor_util")]
+#[test]
+pub fn ed25519()
+{
+    let ee = include_bytes!("ed25519/ee.der");
+    let ca = include_bytes!("ed25519/ca.der");
+
+    let ee_input = untrusted::Input::from(ee);
+    let inter_vec = vec![];
```
Comment (briansmith) [inline tests/integration.rs]: I think `inter_vec` should just be `let intermediates = &[];`

Comment (shahn) [reply tests/integration.rs]: Hopefully addressed, left out a variable for intermediates at all and passing &[] directly into verify_is_valid_tls_server_cert().

**Comment by shahn** [comment]
Thanks, addressed review & pushed new commit with altered commit message.

**Comment by briansmith** [comment]
Thanks! Merged!

**Pull request merged**